### PR TITLE
Allow adding `<extensions>true</extensions>` to plugin settings.

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**
Fixes gh-920

### CODE REVIEW

File: initializr-generator/src/test/java/io/spring/initializr/generator/buildsystem/maven/MavenBuildWriterTests.java [hunk 1]
```diff
@@ -587,4 +548,11 @@ private void generatePom(MavenBuild mavenBuild, Consumer<NodeAssert> consumer)
 		consumer.accept(new NodeAssert(out.toString()));
 	}
 
+	private MavenBuild demoMavenBuild() {
```
Comment (wilkinsona) [inline initializr-generator/src/test/java/io/spring/initializr/generator/buildsystem/maven/MavenBuildWriterTests.java]: We prefer to keep commits as focussed as possible. While certainly not a bad change to make, this is unrelated to adding support for settings `<extensions>true</extensions>`. Could you revert this part of the changes please?

File: initializr-generator/src/main/java/io/spring/initializr/generator/buildsystem/maven/MavenPlugin.java [hunk 2]
```diff
@@ -99,6 +102,17 @@ public Configuration getConfiguration() {
...
 	}
 
+	public boolean shouldLoadExtensions() {
+		return this.loadExtensions;
+	}
+
+	/**
+	 * Enables loading plugin extensions.
+	 */
+	public void loadExtensions() {
```
Comment (wilkinsona) [inline initializr-generator/src/main/java/io/spring/initializr/generator/buildsystem/maven/MavenPlugin.java]: The other methods on `MavenPlugin` that map onto XML tags are generally named after the tag. Along those lines, I wonder if this should be named `extensions` rather than `loadExtensions`.

File: initializr-generator/src/test/java/io/spring/initializr/generator/buildsystem/maven/MavenBuildWriterTests.java (hunk 1 above)
Comment (OlgaMaciaszek) [reply initializr-generator/src/test/java/io/spring/initializr/generator/buildsystem/maven/MavenBuildWriterTests.java]: Ok. I've just stumbled on it and realised I will be repeating the exact same code, so thought would extract it along on the fly. Will revert it.

**Comment by wilkinsona** [comment]
Thanks for the PR and subsequent updates, @OlgaMaciaszek. The changes are now in master.
